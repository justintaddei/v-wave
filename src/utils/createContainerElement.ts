/**
 * Creates a container element for the wave.
 */
export const createContainer = (
  { borderTopLeftRadius, borderTopRightRadius, borderBottomLeftRadius, borderBottomRightRadius }: CSSStyleDeclaration,
  tagName: string
) => {
  const waveContainer = document.createElement(tagName)
  waveContainer.style.top = '0'
  waveContainer.style.left = '0'
  waveContainer.style.width = '100%'
  waveContainer.style.height = '100%'
  waveContainer.style.display = 'block'
  waveContainer.style.position = 'absolute'
  waveContainer.style.borderRadius = `${borderTopLeftRadius} ${borderTopRightRadius} ${borderBottomRightRadius} ${borderBottomLeftRadius}`
  waveContainer.style.overflow = 'hidden'
  waveContainer.style.pointerEvents = 'none'

  // Meet Safari, the new IE ðŸ’©
  waveContainer.style.webkitMaskImage = '-webkit-radial-gradient(white, black)'

  waveContainer.dataset.vWaveContainerInternal = 'true'

  return waveContainer
}
