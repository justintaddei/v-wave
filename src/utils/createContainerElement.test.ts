import { expect, test } from 'vitest'
import { createContainer } from './createContainerElement'

test('createContainerElement returns a an element based on `tagName`', () => {
  expect(createContainer({} as CSSStyleDeclaration, 'div')).toMatchSnapshot()
  expect(createContainer({} as CSSStyleDeclaration, 'span')).toMatchSnapshot()
})
