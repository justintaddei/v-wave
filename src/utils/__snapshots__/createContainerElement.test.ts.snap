// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`createContainerElement returns a an element based on \`tagName\` 1`] = `
<div
  style="top: 0px; left: 0px; width: 100%; height: 100%; display: block; position: absolute; border-radius: undefined undefined undefined undefined; overflow: hidden; pointer-events: none; webkit-mask-image: -webkit-radial-gradient(white, black);"
/>
`;

exports[`createContainerElement returns a an element based on \`tagName\` 2`] = `
<span
  style="top: 0px; left: 0px; width: 100%; height: 100%; display: block; position: absolute; border-radius: undefined undefined undefined undefined; overflow: hidden; pointer-events: none; webkit-mask-image: -webkit-radial-gradient(white, black);"
/>
`;
